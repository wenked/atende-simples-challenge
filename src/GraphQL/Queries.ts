import { gql } from '@apollo/client';

export const VEHICLESNAMES_QUERY = gql`
	query {
		allVehicles {
			vehicles {
				name
				id
			}
		}
	}
`;

export const FILMSTITLES_QUERY = gql`
	query {
		allFilms {
			films {
				title
				id
			}
		}
	}
`;

export const PEOPLENAMES_QUERY = gql`
	query {
		allPeople {
			people {
				id
				name
			}
		}
	}
`;

export const ALLDATA_QUERY = gql`
	query {
		allFilms {
			films {
				episodeID
				title
				openingCrawl
				director
				producers
				releaseDate
				id
			}
		}
		allPeople {
			people {
				id
				name
				birthYear
				eyeColor
				gender
				hairColor
				height
				mass
				skinColor
				homeworld {
					name
				}
				species {
					name
				}
				created
			}
		}
		allVehicles {
			vehicles {
				id
				name
				model
				vehicleClass
				manufacturers
				costInCredits
				length
				crew
				passengers
				maxAtmospheringSpeed
				cargoCapacity
				consumables
			}
		}
	}
`;

export const FILMS_QUERY = gql`
	query {
		allFilms {
			films {
				episodeID
				title
				openingCrawl
				director
				producers
				releaseDate
				created
				edited
				id
			}
		}
	}
`;

export const PEOPLE_QUERY = gql`
	query {
		allPeople {
			people {
				id
				name
				birthYear
				eyeColor
				gender
				hairColor
				height
				mass
				skinColor
				homeworld {
					name
				}
				species {
					name
				}
				created
			}
		}
	}
`;

export const VEHICLES_QUERY = gql`
	query {
		allVehicles {
			vehicles {
				id
				name
				model
				vehicleClass
				manufacturers
				costInCredits
				length
				crew
				passengers
				maxAtmospheringSpeed
				cargoCapacity
				consumables
			}
		}
	}
`;
