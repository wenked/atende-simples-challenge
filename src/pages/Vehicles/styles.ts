import styled from 'styled-components';

export const Container = styled.div`
	padding: 2rem;
	margin: 2rem;
`;
