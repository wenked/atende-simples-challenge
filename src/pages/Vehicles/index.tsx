import React, { useState, useEffect } from 'react';
import { Container } from './styles';
import { useQuery } from '@apollo/client';
import { VEHICLES_QUERY } from '@graphql/Queries';
import { vehicleInterface } from 'src/types';
import Vehicle from '@components/Vehicle';

const Vehicles: React.FC = () => {
	const { error, loading, data } = useQuery(VEHICLES_QUERY);
	const [vehicles, setVehicles] = useState<vehicleInterface[]>();

	useEffect(() => {
		if (data) {
			console.log(data);
			setVehicles(data.allVehicles.vehicles);
		}
	}, [data]);

	return loading ? (
		<h2>Loading...</h2>
	) : (
		<Container>
			{vehicles?.map((vehicle) => (
				<Vehicle key={vehicle.id} vehicle={vehicle} />
			))}
		</Container>
	);
};

export default Vehicles;
