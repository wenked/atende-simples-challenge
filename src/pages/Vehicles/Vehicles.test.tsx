import { MockedProvider } from '@apollo/client/testing';
import { VEHICLESNAMES_QUERY } from '@graphql/Queries';
import { render } from '@testing-library/react';
import { mount, shallow } from 'enzyme';
import React from 'react';
import { act } from 'react-dom/test-utils';
import Vehicles from './index';

const mocks = [
	{
		request: {
			query: VEHICLESNAMES_QUERY,
		},
		result: {
			data: {
				allVehicles: {
					vehicles: [
						{
							id: 'dmVoaWNsZXM6NA==',
							name: 'Sand Crawler',
						},
						{
							id: 'dmVoaWNsZXM6Ng==',
							name: 'T-16 skyhopper',
						},
						{
							id: 'dmVoaWNsZXM6Nw==',
							name: 'X-34 landspeeder',
						},
						{
							id: 'dmVoaWNsZXM6OA==',
							name: 'TIE/LN starfighter',
						},
						{
							id: 'dmVoaWNsZXM6MTQ=',
							name: 'Snowspeeder',
						},
					],
				},
			},
		},
	},
];

describe('rendering component', () => {
	it('rendering without crashing', async () => {
		const wrapper = mount(
			<MockedProvider
				mocks={mocks}
				addTypename={false}
				defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
				<Vehicles />
			</MockedProvider>
		);

		await act(async () => {
			await new Promise((resolve) => setTimeout(resolve, 1000));
		});
		wrapper.update();
		expect(wrapper.find('ItemListContainer').first().text()).toBe(
			'Sand Crawler'
		);
		console.log(wrapper.debug());
	});
});

test('snapshot', async () => {
	const { container } = render(
		<MockedProvider
			mocks={mocks}
			addTypename={false}
			defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
			<Vehicles />
		</MockedProvider>
	);

	await act(async () => {
		await new Promise((resolve) => setTimeout(resolve, 1000));
	});

	expect(container.firstChild).toMatchSnapshot();
});
