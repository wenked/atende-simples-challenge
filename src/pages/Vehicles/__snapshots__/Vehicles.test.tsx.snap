// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="sc-gsTCUz uRvXt"
>
  <div
    class="sc-bdfBwQ vBmnt"
  >
    Sand Crawler
  </div>
  <div
    class="sc-bdfBwQ vBmnt"
  >
    T-16 skyhopper
  </div>
  <div
    class="sc-bdfBwQ vBmnt"
  >
    X-34 landspeeder
  </div>
  <div
    class="sc-bdfBwQ vBmnt"
  >
    TIE/LN starfighter
  </div>
  <div
    class="sc-bdfBwQ vBmnt"
  >
    Snowspeeder
  </div>
</div>
`;
